src/mod_fastcgi.c

```diff
@@ -3552,6 +3593,27 @@ static handler_t fcgi_check_extension(server *srv, connection *con, void *p_d, i
 
 			if (buffer_is_empty(ext->key)) continue;
 
+			/* do not reuse same authorizer, skip if it's in FastCGI-Authorizer env var */
```

Comment:
> What is the use case here? con->loops_per_request test elsewhere will prevent infinite loops.

`con->loops_per_request` will prevent infinite loops, but cancels the request with a 5xx error (in case there is something wrong in the code or the config of authorizers).

> Should the 'continue' be 'break'? Should the check be done once below the loop, only if there is a match?

No, that's intended. Usually, there is only one authorizer, but IMHO it is possible that there is more than one.

The second time the request is handled, we are iterating over all the configured extensions. Without checking the env var, the request would be rescheduled to the same authorizer, which would result in an infinite loop. If we would do a check based on if a flag is set and the extension is for an authorizer (like in the old patch), only one authorizer could be used. Checking the extensions like we do here, we can "jump" over the already used ones (which are listed in the env var) by using `continue`, and select the next one in the array which fullfills the requirements for processing the request. Concluded, this allows multiple authorizers and the last handling of a responder.

> Also, the implementation below only be a few lines of code, but it could probably be more efficient if you also modify the code that inserts FastCGI-Authorizer into the environment. If you start FastCGI-Authorizer with '|' and modify the existing value and always append 'ext|', then you can strstr for '|ext|'.

Wow, that would be much simpler. I only thought about classical header formats, and the multiple value handling came in handy, as it was already there.
I would implement it like this, but I will wait, if you maybe come up with a better solution instead of that env var.
Just as a note: In the case only one authorizer is configured (which should be the most common), only the most simple `strcmp` is used :) So it usually should be efficient, but the implementation of your idea would be much cleaner.

> BTW, I am still mulling over the choice of 'FastCGI-Authorizer' env var name. It will leak into CGI, FastCGI, SCGI, and SSI environments, since lighttpd does not have the equivalent to Apache module 'notes', which do not leak into subprocess environments.

Yeah, I thought the same. Unfornately, I didn't found anything more suitable in `src/base.h#L387`. I don't think it is much of a problem, because the application only can see that there was an authorizer handling the request and for which prefix/extension it was configured.
Anyway, if you would add sth. like a `notes` array to the `connection` struct, I would rewrite the patch to use it. But I'm not sure if that is overkill, as I don't know how that would influence the overall performance, since it will be unused for most of the requests.

Or we could implement it like the HTTP_PROXY filter in `fcgi_env_add_request_headers`, so we don't forward our env var to the processes. But that seems a bit hacky.